src/sys/windows/tcp.rs

```diff
@@ -397,6 +405,10 @@ impl StreamImp {
             self.inner.socket.write_overlapped(&buf[pos..], overlapped)
         };
         match err {
+            Ok(true) if me.instant_notify => {
+                me.write = State::Empty;
+                self.add_readiness(me, Ready::writable());
```

Comment:
yeah, that'll need modification to MIOW though, since we'll need to pass `lpNumberOfBytesSend` to WSASend in `write_overlapped`.

```
A pointer to the number, in bytes, sent by this call if the I/O operation completes immediately.
Use NULL for this parameter if the lpOverlapped parameter is not NULL to avoid potentially erroneous results.
```

Golang and LIBUV seem to use this, which is surprising since the `errorneous results` doesn't sound good.